experiments/conductor/cmd/runner/controller_commands.go

```diff
@@ -464,6 +465,302 @@ func moveTestToSubDir(ctx context.Context, opts *RunnerOptions, branch Branch, e
+    - Only include required fields from the CRD of <kind>.
+    - Follow the schema defined in the CRD file of <kind>.
+    - Do not add any field not defined in the CRD file of <kind>.
+
+Use ReadFile to read the %s/create.yaml file.
+Use ReadFile to read the %s/update.yaml file.
+Use ReadFile to read all the CRD files.
+Check all the spec fields including the nested fields in the %s/create.yaml file and the %s/update.yaml file.
+If any field value in %s/create.yaml or %s/update.yaml ends with "${uniqueId}", then there must be a %s/dependencies.yaml.
+If there is no field name in %s/create.yaml or %s/update.yaml other than "spec.projectRef" ending with "Ref", do nothing and leave.
+Use CreateFile to write the YAML content to the %s/dependencies.yaml file if it doesn't exist; or overwrite the existing file.
+Respond only with the YAML content, no explanations.`,
+			testDir, testDir, testDir, testDir, testDir, branch.Group, branch.Kind, testDir, currentYear, testDir, testDir, testDir, testDir, testDir, testDir, testDir, testDir, testDir, testDir, testDir, testDir)
+
+		cfg = CommandConfig{
+			Name:         "Generate Dependencies YAML",
+			Cmd:          "codebot",
+			Args:         []string{"--prompt=/dev/stdin"},
+			Stdin:        strings.NewReader(dependenciesPrompt),
+			WorkDir:      opts.branchRepoDir,
+			RetryBackoff: GenerativeCommandRetryBackoff,
+		}
+
+		_, err = executeCommand(opts, cfg)
+		if err != nil {
+			return nil, nil, fmt.Errorf("failed to generate dependencies.yaml: %w", err)
+		}
+	}
+
+	return []string{parentDir}, nil, nil
+}
+
+func extractMissingFieldsFromExceptionLine(lines []string) ([]string, error) {
+	result := make([]string, 0)
+	for _, line := range lines {
+		parts := strings.Split(line, "\"")
+		if len(parts) != 3 {
+			return nil, fmt.Errorf("can't understand the format of the missing field exception line")
+		}
+		fieldPath := parts[1]
+		parts = strings.Split(fieldPath, ".")
+		if parts[len(parts)-1] == "external" {
+			log.Printf("Skipping external field %s under reference field", fieldPath)
+			continue
+		}
+		if parts[len(parts)-1] == "value" {
+			log.Printf("Skipping value field %s under sensitive field", fieldPath)
+			continue
+		}
+		result = append(result, fieldPath)
+	}
+	return result, nil
+}
+
+func matchLineInFileByPrefix(path, prefix string) ([]string, error) {
+	result := make([]string, 0)
+	file, err := os.Open(path)
+	if err != nil {
+		return nil, fmt.Errorf("error opening %s: %w", path, err)
+	}
+	defer file.Close()
+
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		line := scanner.Text()
+		line = strings.TrimSpace(line)
+		if strings.HasPrefix(line, prefix) {
+			result = append(result, line)
+		}
+	}
+
+	if err := scanner.Err(); err != nil {
+		return nil, fmt.Errorf("error scanning %s: %w", path, err)
+	}
+	return result, nil
+}
+
+func pluralOf(word string) string {
+	if strings.HasSuffix(word, "f") {
```

Comment:
I think this case is hard to get correct. 
Puff -> puffs not puves
staff -> staff or staves not stafves
brief -> brief not brieves

Maybe not worth special the extra complexity.